service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ResourceExceptionHandler.java

```diff
@@ -38,13 +41,31 @@ public void handleAboutToStartOrSubmitCallbackException(RuntimeException excepti
     public ResponseEntity<AboutToStartOrSubmitCallbackResponse> handleLogAsWarningException(
         LogAsWarningException exception, HttpServletRequest request) {
         log.warn("Ignorable exception for caller {}. {}", getCaller(request), exception.getMessage());
-        
+
         return ResponseEntity.ok(AboutToStartOrSubmitCallbackResponse.builder()
             .errors(List.of(exception.getUserMessage()))
             .build());
     }
 
     private String getCaller(HttpServletRequest request) {
-        return String.format("(id='%s', roles='%s')", request.getHeader("user-id"), request.getHeader("user-roles"));
+        return String.format("(id='%s', roles='%s')", getUserId(request), getUserRoles(request));
+    }
+
+    @SuppressWarnings("EmptyCatchBlock")
+    private String getUserRoles(HttpServletRequest request) {
+        try {
+            return URLEncoder.encode(request.getHeader("user-roles"), StandardCharsets.UTF_8).replaceAll("%2C",",");
+        } catch (Exception e) {
+        }
```

Comment:
maybe instead of the empty blocks that need suppressing, you could just log that we are unable to parse the roles/user id as a warning